src/test_final_models_from_manual_folds_with_custom_loss.py

```diff
@@ -0,0 +1,303 @@
+        "geomean_isf": "ISF",
+        "geomean_weighted_cir": "CIR",
+        "bmi": "BMI",
+        "geomean_total_daily_carbs_in_chunk_outcomes": "CHO",
+        "geomean_total_daily_insulin_dose_in_chunk_outcomes": "TDD",
+    },
+    inplace=True,
+)
+
+# remove any rows where this is a value <= 0
+clean_data = all_data[np.sum(all_data <= 0, axis=1) == 0].copy()
+clean_data.reset_index(drop=True, inplace=True)
+
+clean_data["X_intercept"] = 1
+clean_data["BASAL"] = clean_data["BR"] * 24
+clean_data["log_BASAL"] = np.log(clean_data["BASAL"] + LOG_CONSTANT)
+clean_data["log_BR"] = np.log(clean_data["BR"] + LOG_CONSTANT)
+clean_data["log_ISF"] = np.log(clean_data["ISF"] + LOG_CONSTANT)
+clean_data["log_CIR"] = np.log(clean_data["CIR"] + LOG_CONSTANT)
+clean_data["log_BMI"] = np.log(clean_data["BMI"] + LOG_CONSTANT)
+clean_data["log_CHO"] = np.log(clean_data["CHO"] + LOG_CONSTANT)
+clean_data["log_TDD"] = np.log(clean_data["TDD"] + LOG_CONSTANT)
+clean_data["1/TDD"] = 1 / clean_data["TDD"]
+
+y_cols = ["BASAL", "log_BASAL", "ISF", "log_ISF", "CIR", "log_CIR"]
+x_cols = ["X_intercept", "BMI", "log_BMI", "CHO", "log_CHO", "TDD", "log_TDD", "1/TDD"]
+
+X = clean_data[x_cols]
+y = clean_data[y_cols]
+
+# break the data into 70% train and 30% test set
+X_train, X_test, y_train, y_test = train_test_split(
+    X,
+    y,
+    test_size=0.3,
+    random_state=1,
+)
+
+# %% START OF NEW CODE
+keep_columns = [
+    "y",
+    "X_intercept",
+    "BMI",
+    "CHO",
+    "TDD",
+    "beta_X_intercept",
+    "beta_BMI",
+    "beta_log_BMI",
+    "beta_CHO",
+    "beta_log_CHO",
+    "beta_TDD",
+    "beta_log_TDD",
+    "beta_1/TDD",
+    "val_loss",
+]
+
+beta_columns = [
+    "beta_X_intercept",
+    "beta_BMI",
+    "beta_log_BMI",
+    "beta_CHO",
+    "beta_log_CHO",
+    "beta_TDD",
+    "beta_log_TDD",
+    "beta_1/TDD",
+]
+
+x_columns = ["X_intercept", "BMI", "CHO", "TDD"]
+
+
+df = pd.read_csv(file_name)
+
+# only keep models that have a cross validation model score (val_loss)
+converged_model_mask = df["val_loss"].notnull()
+# only keep models that have beta_coefficients not equal to 0
+non_zero_coefficients_mask = np.sum(df[beta_columns] == 0, axis=1) == 0
+
+final_equation_mask = converged_model_mask & non_zero_coefficients_mask
+final_equations_df = df.loc[final_equation_mask, keep_columns].copy().reset_index(drop=True)
+
+if "BASAL" in file_name:
+    final_equations_df.loc[
+        final_equations_df.index.max() + 1, ["y", "X_intercept", "BMI", "CHO", "TDD", "beta_TDD"]
+    ] = ["BASAL", "off", "off", "off", "TDD", 0.5]
+
+elif "ISF" in file_name:
+    final_equations_df.loc[
+        final_equations_df.index.max() + 1, ["y", "X_intercept", "BMI", "CHO", "TDD", "beta_1/TDD"]
+    ] = ["ISF", "off", "off", "off", "1/TDD", 1700]
+
+elif "CIR" in file_name:
+    final_equations_df.loc[
+        final_equations_df.index.max() + 1, ["y", "X_intercept", "BMI", "CHO", "TDD", "beta_1/TDD"]
+    ] = ["CIR", "off", "off", "off", "1/TDD", 450]
```

Comment:
This is to add the standard AACE equation forms to the analysis, right?

Reply:
yes, that is correct. I added a comment in the code.